metagraph/src/common/caches.hpp

```diff
@@ -0,0 +1,80 @@
+#ifndef __MTG_CACHES__
+#define __MTG_CACHES__
+
+#include <list>
+#include <optional>
+#include <mutex>
+#include <shared_mutex>
+#include <utility>
+
+#include <tsl/hopscotch_map.h>
+
+namespace mtg {
+namespace common {
+
+// A fast LRU cache
+template <typename Key, typename Value>
+class LRUCache {
```

Comment:
Is this faster than the stuff in our external library?

Reply:
For this case, it actually is...at some point we should open up a new PR to replace the external library with this